src/zimscraperlib/zim/metadata.py

```diff
@@ -1,145 +1,471 @@
+        super().__init__(name="Description", value=value)
+
+    @property
+    def libzim_value(self) -> bytes:
+        """The value to pass to the libzim for creating the metadata"""
+        value = super().libzim_value
+        if check_metadata_conventions:
+            if nb_grapheme_for(value.decode()) > MAXIMUM_DESCRIPTION_METADATA_LENGTH:
+                raise ValueError("Description is too long.")
+        return value
+
+
+class LongDescriptionMetadata(_MandatoryTextMetadata):
+    """The LongDescription metadata"""
+
+    def __init__(self, value: bytes | io.BytesIO | str) -> None:
+        super().__init__(name="LongDescription", value=value)
+
+    @property
+    def libzim_value(self) -> bytes:
+        """The value to pass to the libzim for creating the metadata"""
+        value = super().libzim_value
+        if check_metadata_conventions:
+            if (
+                nb_grapheme_for(value.decode())
+                > MAXIMUM_LONG_DESCRIPTION_METADATA_LENGTH
+            ):
+                raise ValueError("LongDescription is too long.")
+        return value
+
+
+class DateMetadata(_TextMetadata):
+    """The Date metadata"""
+
+    def __init__(self, value: bytes | str | datetime.date | datetime.datetime) -> None:
+        if not isinstance(value, bytes | str | datetime.date | datetime.datetime):
+            raise ValueError(
+                f"Unexpected type passed to {self.__class__.__qualname__}: "
+                f"{value.__class__.__qualname__}"
+            )
+        super().__init__(
+            name="Date",
+            value=(
+                value
+                if isinstance(value, bytes)
+                else (
+                    value.encode()
+                    if isinstance(value, str)
+                    else value.strftime("%Y-%m-%d").encode()
+                )
             ),
+        )
+        self.value = value
+        _ = self.libzim_value  # to check for errors
+
+    @property
+    def libzim_value(self) -> bytes:
+        """The value to pass to the libzim for creating the metadata"""
+        if isinstance(self.value, datetime.date | datetime.datetime):
+            return self.value.strftime("%Y-%m-%d").encode()
+        match = re.match(
+            r"(?P<year>\d{4})-(?P<month>\d{2})-(?P<day>\d{2})",
+            self.value.decode() if isinstance(self.value, bytes) else self.value,
+        )
+        if not match:
+            raise ValueError("Invalid date format, not matching regex yyyy-mm-dd")
+        try:
+            datetime.date(**{k: int(v) for k, v in match.groupdict().items()})
+        except Exception as exc:
+            raise ValueError(f"Invalid date format: {exc}") from None
+        return self.value if isinstance(self.value, bytes) else self.value.encode()
+
+
+class IllustrationMetadata(_MandatoryMetadata):
+    """Any Illustration_**x**@* metadata"""
+
+    def __init__(self, name: str, value: bytes | io.BytesIO) -> None:
```

Comment:
Given Illustrations have a static name pattern, maybe we should not ask for a name but sizes and scale instead. That would remove the need for the pattern check.
We can imagine having a special `Default48IllustrationMetadata` or something that only takes data and supers this with 48x48 at 1 ; since that's mandatory